qcodes/dataset/sqlite_base.py

```diff
@@ -1350,10 +1368,13 @@ def get_parameter_data(conn: ConnectionPlus,
 
         # if we have array type parameters expand all other parameters
         # to arrays
-        if 'array' in types and ('numeric' in types or 'text' in types):
+        if 'array' in types and ('numeric' in types or 'text' in types
+                                 or 'complex in types'):
```

Comment:
Looks like typo. Is it caught by any tests? what kind of tests would catch this?
```suggestion
                                 or 'complex' in types):
```

Reply:
As I understand it we cannot write a test that would catch this. The non empty string always evaluates to True. However removing the the `and ...` part of the if statement will only result in a slight performance degradation so there is no way of testing for it

Reply:
exactly, i got similar conclusions as well ((

anyway, this is yet another example that confirms that "code review" is important and necessary for ensuring code quality blablablabla.